modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -0,0 +1,442 @@
+        // import JsonCreator if JsonProperty is imported
+        // used later in recursive import in postProcessingModels
+        importMapping.put("com.fasterxml.jackson.annotation.JsonProperty", "com.fasterxml.jackson.annotation.JsonCreator");
+
+        // TODO when adding invokerPackage
+        //importMapping.put("StringUtil", invokerPackage + ".StringUtil");
+
+        if (!additionalProperties.containsKey(CodegenConstants.LIBRARY)) {
+            additionalProperties.put(CodegenConstants.LIBRARY, library);
+        }
+
+        if (additionalProperties.containsKey(BASE_PACKAGE)) {
+            this.setBasePackage((String) additionalProperties.get(BASE_PACKAGE));
+        } else {
+            additionalProperties.put(BASE_PACKAGE, basePackage);
+        }
+
+        if (additionalProperties.containsKey(SERVER_PORT)) {
+            this.setServerPort((String) additionalProperties.get(SERVER_PORT));
+        } else {
+            additionalProperties.put(SERVER_PORT, serverPort);
+        }
+
+        if (additionalProperties.containsKey(EXCEPTION_HANDLER)) {
+            this.setExceptionHandler(Boolean.valueOf(additionalProperties.get(EXCEPTION_HANDLER).toString()));
+        }
+        writePropertyBack(EXCEPTION_HANDLER, exceptionHandler);
+
+        if (additionalProperties.containsKey(GRADLE_BUILD_FILE)) {
+            this.setGradleBuildFile(Boolean.valueOf(additionalProperties.get(GRADLE_BUILD_FILE).toString()));
+        }
+        writePropertyBack(GRADLE_BUILD_FILE, gradleBuildFile);
+
+        if (additionalProperties.containsKey(SWAGGER_ANNOTATIONS)) {
+            this.setSwaggerAnnotations(Boolean.valueOf(additionalProperties.get(SWAGGER_ANNOTATIONS).toString()));
+        }
+        writePropertyBack(SWAGGER_ANNOTATIONS, swaggerAnnotations);
+
+        if (additionalProperties.containsKey(SERVICE_INTERFACE)) {
+            this.setServiceInterface(Boolean.valueOf(additionalProperties.get(SERVICE_INTERFACE).toString()));
+        }
+        writePropertyBack(SERVICE_INTERFACE, serviceInterface);
+
+        if (additionalProperties.containsKey(SERVICE_IMPLEMENTATION)) {
+            this.setServiceImplementation(Boolean.valueOf(additionalProperties.get(SERVICE_IMPLEMENTATION).toString()));
+        }
+        writePropertyBack(SERVICE_IMPLEMENTATION, serviceImplementation);
+
+        if (additionalProperties.containsKey(USE_BEANVALIDATION)) {
+            this.setUseBeanValidation(convertPropertyToBoolean(USE_BEANVALIDATION));
+        }
+        writePropertyBack(USE_BEANVALIDATION, useBeanValidation);
+
+        modelTemplateFiles.put("model.mustache", ".kt");
+        apiTemplateFiles.put("api.mustache", ".kt");
+        supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
+
+        if (this.serviceInterface) {
+            apiTemplateFiles.put("service.mustache", "Service.kt");
+        } else if (this.serviceImplementation) {
+            additionalProperties.put(SERVICE_INTERFACE, true);
```

Comment:
My thinking was that you can't generate the implementation classes with missing interfaces, as it would lead to compile errors due to the missing generated interfaces.

I will however re-visit this logic, and perhaps add appropriate logging. Or perhaps if `serviceInterface` is set to `false`, just generate the standalone classes that don't implement anything.